data/deduplicator/delegate/delegate.go

```diff
@@ -13,25 +13,30 @@ package delegate
 import (
 	"github.com/tidepool-org/platform/app"
 	"github.com/tidepool-org/platform/data/deduplicator"
-	"github.com/tidepool-org/platform/data/deduplicator/truncate"
 	"github.com/tidepool-org/platform/data/store"
 	"github.com/tidepool-org/platform/data/types/base/upload"
 	"github.com/tidepool-org/platform/log"
 )
 
-func NewFactory() deduplicator.Factory {
-	return &factory{
-		[]deduplicator.Factory{
-			truncate.NewFactory(),
-		},
-	}
-}
-
 type factory struct {
 	factories []deduplicator.Factory
 }
 
+func NewFactory(factories []deduplicator.Factory) (deduplicator.Factory, error) {
+	if len(factories) == 0 {
+		return nil, app.Error("delegate", "factories is missing")
```

Comment:
Yes, I actually had two separate checks `factories == nil` and `len(factories) == 0`, but decided they were essentially the same thing (ie. no factories provided) so I just combined the two.